### ISSUE TITLE
Equivalent to pytorch tril / masked_fill

### ISSUE THREAD

**Issue by okpatil4u** [huggingface/candle#485]
How would you implement following pytorch code into candle ?

```
wei = torch.ones(T,T)
tril = torch.tril(torch.ones(T,T))
wei = wei.masked_fill(tril==0, float('-inf'))
wei = F.softmax(wei, dim=-1)
out = wei @ x
```

**Comment by LaurentMazare** [comment]
You'll probably find plenty of related code in the `candle-examples` directory, e.g. for llama the masked-fill bit is done [here](https://github.com/huggingface/candle/blob/d32e8199cd6c8381aa309528675d6d6a88c0f850/candle-examples/examples/llama/model.rs#L325) and you will find the `tril`, `softmax`, etc in the same file (for `tril` the tensor is built manually at the moment).

**Comment by okpatil4u** [comment]
Got it. This will need some work.

**Comment by okpatil4u** [comment]
Also, is there an equivalent function to **torch.ones** ?  This may be hurting me the most.

**Comment by LaurentMazare** [comment]
[Tensor::ones](https://docs.rs/candle-core/latest/candle_core/struct.Tensor.html#method.ones) ?

**Comment by okpatil4u** [comment]
Apologies, I missed it. Thanks for pointing this out.
